lib/ffmpeg/transcoder.rb

```diff
@@ -95,7 +105,13 @@ def pre_encode_if_necessary
       @movie.unescaped_paths.each_with_index do |path, index|
         audio_map = determine_audio_for_pre_encode(path)
 
-        command = "#{@movie.ffmpeg_command} -y -i #{Shellwords.escape(path)} -movflags faststart #{pre_encode_options} -r #{output_frame_rate} -filter_complex \"[0:v]scale=#{max_width}:#{max_height}:force_original_aspect_ratio=decrease,pad=#{max_width}:#{max_height}:(ow-iw)/2:(oh-ih)/2:color=black,setsar=1[Scaled]\" -map \"[Scaled]\" #{audio_map} #{@movie.interim_paths[index]}"
+        # Only re-scale the video if there are multiple inputs. If a single input video contains dynamic resolution it can cause issues with the filter_complex filter
+        complex_filter = ""
+        if @movie.paths.size > 1
```

Comment:
This is no longer using the calculated max_width & max_height -- is that intentional?

Reply:
It still is if this evaluates to true.
It doesn't seem the pre-encode needs it for when this is a single input trim as ffmpeg sizes it to the `-s` argument on its own

Reply:
And when a video contains multiple resolutions, does it correctly not squish/stretch the larger resolution portion?

Reply:
It is correct!
Heres 3 trims stitched together in various orders. (Stitching trim + trim + trim has no issues)
https://staging-aquaman.vidyard.com/organizations/982359/players/ZXFveGaFPaUe7dSW6TsXDe?edit=true
https://staging-aquaman.vidyard.com/organizations/982359/players/E6AwKXKyTpWRU4kVjt4RcV?edit=true
https://staging-aquaman.vidyard.com/organizations/982359/players/7oEWEuuNeq1t6W9C7QKbZJ?edit=true

Heres an additional 4 variations of stitches
https://staging-aquaman.vidyard.com/organizations/982359/players/yqxDfCKFiuDCsWth9zTgP4?edit=true
https://staging-aquaman.vidyard.com/organizations/982359/players/iTNmJPZPTBM8mEdoDbWTMu?edit=true
https://staging-aquaman.vidyard.com/organizations/982359/players/RtmZKzS7UGQGyzsJMU6PSk?edit=true
https://staging-aquaman.vidyard.com/organizations/982359/players/HkDmFS3dvJBYpDYPer6vuv?edit=true